test/RelayClient.test.ts

```diff
@@ -146,18 +153,84 @@ describe('RelayClient', () => {
                 relayClient.validateSmartWallet(fakeTransactionDetails),
                 error.message
             );
         });
+    });
 
-        it('should fail if tansactionDetails is null', async () => {
-            await assert.isRejected(
-                relayClient.validateSmartWallet(null),
-                "Cannot read properties of null (reading 'callForwarder')"
+    describe('estimateGasLimit', function () {
+        const responseRelayEstimation: RelayEstimation = {
+            estimation: '151800',
+            exchangeRate: '0.000003323449073',
+            gasPrice: '60000000',
+            requiredTokenAmount: '2.74053e18',
+            requiredNativeAmount: '0.00000311892'
+        };
+
+        const transactionDetails: EnvelopingTransactionDetails = {
+            from: '0x1',
+            to: constants.ZERO_ADDRESS,
+            value: '0',
+            callForwarder: '0x2',
+            data: '0x0'
+        };
+
+        const pingResponse = {
+            feesReceiver: '0x3'
+        } as PingResponse;
+
+        beforeEach(function () {
+            httpClient = createStubInstance(HttpClient, {
+                estimateGasLimit: Promise.resolve(responseRelayEstimation),
+                getPingResponse: Promise.resolve(pingResponse)
+            });
+            contractInteractor = createStubInstance(ContractInteractor);
+            accountManager = createStubInstance(AccountManager);
+            mockDependencies = {
+                httpClient,
+                contractInteractor,
+                knownRelaysManager: new KnownRelaysManager(
+                    contractInteractor,
+                    defaultConfig,
+                    EmptyFilter
+                ),
+                accountManager,
+                transactionValidator: new RelayedTransactionValidator(
+                    contractInteractor,
+                    defaultConfig
+                ),
+                pingFilter: GasPricePingFilter,
+                relayFilter: EmptyFilter,
+                scoreCalculator: DefaultRelayScore,
+                config: defaultConfig
+            };
+            replace(
+                configurator,
+                'getDependencies',
+                fake.returns(mockDependencies)
+            );
+            mockHttpProvider = stubInterface<HttpProvider>();
+            relayClient = new RelayClient(mockHttpProvider, defaultConfig);
+            replace(
+                relayClient,
+                'getInternalCallCost',
+                fake.returns(Promise.resolve(5))
+            );
+        });
+
+        afterEach(function () {
+            restore();
+        });
+
+        it('should estimate relay gas limit', async function () {
+            const estimation = await relayClient.estimateGasLimit(
+                transactionDetails,
+                true
             );
+            expect(estimation).to.be.equal(responseRelayEstimation);
         });
```

Comment:
I think we should add a test to cover different scenarios like:
- without signature
- deployment vs relay
- http client returns an error

Reply:
Im including the tests, but I did not do it since this scenarios are tested in the gasEstimator and I thought that this type of scenarios will not be testing what this method actually do, that is basically build the request.